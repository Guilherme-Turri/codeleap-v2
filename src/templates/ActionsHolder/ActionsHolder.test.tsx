import { pageRender } from '../../render/pageRender'
import { screen, fireEvent } from '@testing-library/react'
import { ActionsHolder } from './'

describe('<ActionsHolder />', () => {
  const ActionsHolderProps = {
    handleDelete: jest.fn(),
    handleEdit: jest.fn(),
  }

  it('should render on screen properly', () => {
    const { container } = pageRender(<ActionsHolder {...ActionsHolderProps} />)
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  })

  it('should render on screen properly and call handleDelete fn', () => {
    const { container } = pageRender(<ActionsHolder {...ActionsHolderProps} />)
    expect(container).toBeInTheDocument();
    const deletebtn = screen.getByTestId('delete')
    fireEvent.click(deletebtn);
    expect(ActionsHolderProps.handleDelete).toHaveBeenCalled();
  })

  it('should render on screen properly and call handleEdit fn', () => {
    const { container } = pageRender(<ActionsHolder {...ActionsHolderProps} />)
    expect(container).toBeInTheDocument();
    const editbtn = screen.getByTestId('edit')
    fireEvent.click(editbtn);
    expect(ActionsHolderProps.handleEdit).toHaveBeenCalled();
  })
})
