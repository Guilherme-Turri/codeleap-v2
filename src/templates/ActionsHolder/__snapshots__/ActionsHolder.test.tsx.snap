// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionsHolder /> should render on screen properly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 50px;
  height: 50px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  margin: 0.313rem;
}

.c3 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c2 {
  color: #00CBA9;
  width: 100%;
  height: 100%;
}

.c4 {
  color: #00CBA9;
  width: 100%;
  height: 100%;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      data-testid="edit"
    >
      <div
        class="c2"
        data-testid="Edit"
      >
        <svg
          aria-hidden="true"
          class="c3"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 2H4c-1.103 0-2 .897-2 2v12c0 1.103.897 2 2 2h3v3.767L13.277 18H20c1.103 0 2-.897 2-2V4c0-1.103-.897-2-2-2zm0 14h-7.277L9 18.233V16H4V4h16v12z"
          />
          <path
            d="m13.803 9.189-1.399-1.398-3.869 3.864v1.399h1.399zm.327-3.123 1.398 1.399-1.066 1.066-1.399-1.398z"
          />
        </svg>
      </div>
    </div>
    <div
      class="c1"
      data-testid="delete"
    >
      <div
        class="c4"
        data-testid="Delete"
      >
        <svg
          aria-hidden="true"
          class="c3"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12 1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
